### ISSUE TITLE
Change log timestamp callLater thing to be attached to factory instead of module

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2386]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#2386|
|Type|enhancement|
|Created|2007-01-11 14:18:05Z|
|Branch|https://github.com/twisted/twisted/tree/remove-global-log-2386-2|

The twisted.web speed optimization for timestamps could be made more trial-friendly by making it run by server.Site, rather than on import.

Attachments:

* [issue.2386.fix.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/40c/40cb7530c53bb63ec455a175841921a3af351c0f/4ccaab9714ef1adbab1c76868078a01013e730b9.diff) (6524 bytes) - added by steiza on 2011-03-14 16:31:59Z - Proposed fix
* [issue.2386.fix.2.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/40c/40cb7530c53bb63ec455a175841921a3af351c0f/3c4622d0e927b53ffc18410c358f4e6ded6e4f54.diff) (6630 bytes) - added by steiza on 2011-03-14 20:01:28Z - Updated some documentation, definitely needs review
* [issue.2386.fix.3.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/40c/40cb7530c53bb63ec455a175841921a3af351c0f/6d792d3db4e4d5c21b2c233f79f905b28514b157.diff) (7096 bytes) - added by steiza on 2011-03-14 20:16:46Z - New diff

<details><summary>Searchable metadata</summary>

```
trac-id__2386 2386
type__enhancement enhancement
reporter__itamarst itamarst
priority__normal normal
milestone__ 
branch__branches_remove_global_log_2386_2 branches/remove-global-log-2386-2
branch_author__exarkun__lvh exarkun, lvh
status__closed closed
resolution__fixed fixed
component__web web
keywords__easy easy
time__1168525085000000 1168525085000000
changetime__1300487733000000 1300487733000000
version__None None
owner__lvh lvh
cc__ghazel cc__lvh
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

[#2447](https://github.com/twisted/twisted/issues/2447) is a duplicate of this, and includes a test case.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

Nevermind, [#2447](https://github.com/twisted/twisted/issues/2447) is about DelayedCalls in web code, but not _this_ DelayedCall.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

The timer can be started in the factory's `startFactory` and it can be cleaned up in `stopFactory`.

**Comment by twisted-trac** [comment]
|<img alt="steiza's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|steiza set owner to steiza|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)|@lvh set owner to @lvh<br>@lvh set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_8"></a>|@lvh commented|
|-|-|

(In [31034]) Branching to 'remove-global-log-2386'

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_9"></a>|@lvh commented|
|-|-|

Builds: [http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/remove-global-log-2386]

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_10"></a>|@lvh set status to `new`|
|-|-|

Okay, looks good to me, happy buildbots incoming and as soon as all of them are in I'll merge it :)

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_11"></a>|@lvh set owner to steiza|
|-|-|

Err, except not really. Could you add a docstring explaining the new (private) attributes?

**Comment by twisted-trac** [comment]
|<img alt="steiza's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|steiza removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_13"></a>|@lvh set owner to steiza|
|-|-|

Okay on the method docstring but missing attribute docstrings.

**Comment by twisted-trac** [comment]
|<img alt="steiza's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|steiza removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_15"></a>|@lvh set owner to @lvh<br>@lvh set status to `assigned`|
|-|-|

Looking good! In the future please use more complete sentences, but I'll do it for you this time since otherwise I'm earning gajillions of points off of you ;-)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_16"></a>|@exarkun commented|
|-|-|

(In [31236]) Branching to 'remove-global-log-2386-2'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_17"></a>|@exarkun commented|
|-|-|

(In [31237]) Apply issue.2386.fix.3.diff 

refs [#2386](https://github.com/twisted/twisted/issues/2386)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_18"></a>|@exarkun commented|
|-|-|

(In [31238]) Rename _logDateTimeCall{back,}; insert some whitespace and do some docstring stuff

refs [#2386](https://github.com/twisted/twisted/issues/2386)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_19"></a>|@exarkun commented|
|-|-|

I made a few more tweaks, just docstrings and whitespace and one attribute name.  [Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/remove-global-log-2386-2) look good, so I'll merge this now.  Thanks steiza and lvh!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_20"></a>|@exarkun set status to `closed`|
|-|-|

(In [31240]) Merge remove-global-log-2386-2

Author: steiza
Reviewer: lvh, exarkun
Fixes: [#2386](https://github.com/twisted/twisted/issues/2386)

Move the global timestamp cache from `twisted.web.http` into per-instance
state on `HTTPFactory`.
